### PR TITLE
Optimization of the "PrimitiveAdvText" class across various methods 

### PR DESCRIPTION
**Pull request by manufino**
.. and improved the coding styles

### CODE REVIEW

**Comment by manufino** [comment]
Fixed an issue in PrimitiveAdvText that prevented the selection of mirrored text

**Comment by manufino** [comment]
Improved the intersects method in the PrimitiveOval class

**Comment by manufino** [comment]
I thoroughly tested all the primitives with the two selection methods, and they now seem to be quite efficient. The only thing left is to implement the macros, but that's a bit of a separate task.

File: src/fidocadj/primitives/PrimitiveAdvText.java [hunk 1]
```diff
@@ -465,199 +490,182 @@ public int getDistanceToPoint(int px, int py)
...
-        }
+            if (param instanceof String) {
+                txt = (String) param;
+            } else {
+                if (param instanceof LayerInfo) {
+                    setLayer(((LayerInfo) param).getLayer());
+                } else {
+                    if (param instanceof Integer) {
+                        switch (i) {
+                            case 2 ->
```
Comment (DarwinNE) [inline src/fidocadj/primitives/PrimitiveAdvText.java:575]: There is an error here? Is the -> correct? This does not look as a lambda?

Comment (manufino) [reply src/fidocadj/primitives/PrimitiveAdvText.java:575]: > There is an error here? Is the -> correct? This does not look as a lambda?

Feature introduced in Java 12 as part of a new syntax for switch statements. This new syntax, called "switch expression," allows the switch construct to be used more concisely and with less boilerplate code compared to the traditional syntax.

[See this reference](https://blogs.oracle.com/javamagazine/post/new-switch-expressions-in-java-12)

**Pull request merged**